### PR TITLE
RavenDB-20308 Use Icon component instead of <i> tag

### PR DESCRIPTION
**Pull request by mateuszbartosik**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-20308/Icons-has-wrong-padding

### Additional description
Changed all the <i> usages into <Icon> component

### Type of change
- Optimization

### How risky is the change?
- Low

### Backward compatibility
- Not relevant

### Is it platform specific issue?
- No

### Documentation update
- No documentation update is needed 

### Testing by Contributor
- It has been verified by manual testing

### Testing by RavenDB QA team
- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexToolbarActions.tsx [hunk 1]
```diff
@@ -70,7 +71,7 @@ export default function IndexToolbarAction(props: IndexToolbarActionProps) {
                     onClick={deleteSelectedIndexes}
                     className="margin-right-xxs"
                 >
-                    <i className="icon-trash" />
+                    <Icon icon="trash" className="me-1"></Icon>
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexToolbarActions.tsx]: <Icon />

File: src/Raven.Studio/typescript/components/pages/database/tasks/panels/OngoingEtlTaskDistribution.tsx [hunk 2]
```diff
@@ -83,17 +84,17 @@ export function OngoingEtlTaskDistribution(props: OngoingEtlTaskDistributionProp
...
-                        <i className="icon-connected" /> Status
+                        <Icon icon="connected" className="me-1" /> Status
                     </div>
                     <div>
-                        <i className="icon-warning" /> Error
+                        <Icon icon="warning" className="me-1" /> Error
                     </div>
                     <div>
-                        <i />
+                        <Icon icon="" className="me-1" />
```
Comment (kwiato) [inline src/Raven.Studio/typescript/components/pages/database/tasks/panels/OngoingEtlTaskDistribution.tsx]: Why is this empty?

File: src/Raven.Studio/typescript/components/pages/database/tasks/panels/SubscriptionTaskDistribution.tsx [hunk 3]
```diff
@@ -65,17 +66,17 @@ export function SubscriptionTaskDistribution(props: OngoingEtlTaskDistributionPr
...
-                    <i className="icon-connected" /> Status
+                    <Icon icon="connected" className="me-1" /> Status
                 </div>
                 <div>
-                    <i className="icon-warning" /> Error
+                    <Icon icon="warning" className="me-1" /> Error
                 </div>
                 <div>
-                    <i />
+                    <Icon icon="" className="me-1" />
```
Comment (kwiato) [inline src/Raven.Studio/typescript/components/pages/database/tasks/panels/SubscriptionTaskDistribution.tsx]: empty?

File: src/Raven.Studio/typescript/components/pages/resources/databases/partials/DatabaseDistribution.tsx [hunk 4]
```diff
@@ -28,29 +29,29 @@ export function DatabaseDistribution(props: DatabaseDistributionProps) {
                 <div className="top"></div>
                 {sharded && (
                     <div className="node">
-                        <i className="icon-node" /> Node
+                        <Icon icon="node" className="me-1"></Icon> Node
```
Comment (kwiato) [inline src/Raven.Studio/typescript/components/pages/resources/databases/partials/DatabaseDistribution.tsx]: <Icon></icon> => <Icon />

File: src/Raven.Studio/typescript/components/pages/database/tasks/panels/OngoingEtlTaskDistribution.tsx (hunk 2 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/pages/database/tasks/panels/OngoingEtlTaskDistribution.tsx]: It was empty all along. I've changed it to the bell icon:
image

File: src/Raven.Studio/typescript/components/common/DatabaseGroup.tsx [hunk 5]
```diff
@@ -76,11 +78,11 @@ export function DatabaseGroupType(props: DatabaseGroupTypeProps) {
 
         switch (type) {
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/DatabaseGroup.tsx:79]: please check what type this method returns? string or IconName?

File: src/Raven.Studio/typescript/components/common/Icon.tsx [hunk 6]
```diff
@@ -1,17 +1,20 @@
...
 }
 
 export function Icon(props: IconProps) {
-    const { icon, addon, color, className, ...rest } = props;
+    const { icon, addon, color, margin, className, ...rest } = props;
     const iconClasses = "icon-" + icon;
     const addonClasses = addon ? "icon-addon-" + addon : null;
     const colorClasses = color ? "text-" + color : null;
-    return <i className={classNames(iconClasses, addonClasses, colorClasses, className)} {...rest} />;
+    const marginClass = margin ? margin : "me-1";
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/Icon.tsx]: const marginClass = margin ??  "me-1";

File: src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx [hunk 7]
```diff
@@ -74,9 +74,9 @@ export function MultiCheckboxToggle<T extends string | number = string>({
                         >
                             <div className="label-span">
                                 <div className="label-select-all">
-                                    {selectAllLabel ? selectAllLabel : <Icon icon="accept" />}
+                                    {selectAllLabel ? selectAllLabel : <Icon icon="accept" margin="m-0" />}
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/MultiCheckboxToggle.tsx]: use `??`

File: src/Raven.Studio/typescript/components/common/NodeSet.tsx [hunk 8]
```diff
@@ -41,23 +43,23 @@ interface NodeSetItemProps {
 export function NodeSetLabel(props: NodeSetItemProps) {
     const { children, icon, color, ...rest } = props;
 
-    const colorClass = color ? "text-" + color : undefined;
+    const colorClass = color ? color : undefined;
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/NodeSet.tsx]: ??

File: src/Raven.Studio/typescript/components/common/NodeSet.tsx [hunk 9]
```diff
@@ -41,23 +43,23 @@ interface NodeSetItemProps {
...
+            {icon && <Icon icon={icon} color={colorClass} />}
             <strong className="node-set-label-name">{children}</strong>
         </div>
     );
 }
 
 export function NodeSetItem(props: NodeSetItemProps) {
     const { children, icon, color, extraIconClassName, ...rest } = props;
-    const colorClass = color ? "text-" + color : undefined;
+    const colorClass = color ? color : undefined;
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/NodeSet.tsx]: ??

File: src/Raven.Studio/typescript/components/common/DatabaseGroup.tsx (hunk 5 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/common/DatabaseGroup.tsx:79]: Now it returns `IconName`

File: src/Raven.Studio/typescript/components/common/Icon.tsx (hunk 6 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/common/Icon.tsx]: Changed

File: src/Raven.Studio/typescript/components/common/NodeSet.tsx (hunk 8 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/common/NodeSet.tsx]: Changed

File: src/Raven.Studio/typescript/components/common/NodeSet.tsx (hunk 9 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/common/NodeSet.tsx]: Changed

**Review by ml054 — APPROVED** [review]

**Pull request merged**